gems/aws-sdk-core/spec/aws/credential_resolution_chain_spec.rb

```diff
@@ -69,6 +69,11 @@ module Aws
         expect(client.config.credentials.access_key_id).to eq("ACCESS_KEY_SC1")
       end
 
+      it 'prefers process credentials over metadata credentials' do
```

Comment:
Last bit here, I would also add some tests with this same profile where we ensure that static profile credentials, for example, are pulled in first. It's a bit thorough, but important for regression.